DataFormats/EgammaCandidates/interface/GsfElectron.h

```diff
@@ -183,12 +183,15 @@ class GsfElectron : public RecoCandidate
     // forward core methods
     SuperClusterRef superCluster() const override { return core()->superCluster() ; }
     GsfTrackRef gsfTrack() const override { return core()->gsfTrack() ; }
-    virtual TrackRef closestTrack() const { return core()->ctfTrack() ; }
+    //so it turns out closestCtfTrackRef is overriden by pat::Electron but not closestTrack()
+    //this results in surprising behaviour, ie two functions which looked identical 
+    //would give different results so now closestTrack just calls closestCtfTrackRef
```

Comment:
I'm not sure this comment is appropriate. We should not tune up the base class interface/implementations to the decisions/shortcomings of the derived classes.

Both the ```closestTrack``` and ```closestCtfTrackRef``` methods are first appearing in  the ```GsfElectron```.
It seems like your preference is to have them the same.
If the design is for both to be the same; perhaps better remove one and leave just one to avoid confusion.
Edits in other parts of this PR are suggestive that the ```closestTrack``` should be removed.

Reply:
There is a program of work planned for 11_X to fix this and other things like this, now is not the time.

Reply:
this seems simple enough to be done now; considering that it's fractionally already done

Reply:
I'm more thinking about user code. I would prefer not to make code which compiles with 102X not compile with 106X, particularly when we want to clean this & reco::Photon up in 11_X.